scripts/src/public/services/orders.ts

```diff
@@ -192,26 +229,36 @@ export async function getOrderHistory(
 	};
 }
 
-function orderDbToApi(order: db.MarketplaceOrder): Order {
+function orderDbToApi(order: db.ExternalOrder): ExternalOrder;
+function orderDbToApi(order: db.MarketplaceOrder): MarketplaceOrder;
+function orderDbToApi(order: db.ExternalOrder | db.MarketplaceOrder): ExternalOrder | MarketplaceOrder {
 	if (order.status === "opened") {
 		throw new Error("opened orders should not be returned");
 	}
 
-	return {
+	const base = {
 		id: order.id,
 		offer_id: order.offerId,
 		status: order.status,
-		result: order.value,
 		error: order.error,
 		completion_date: (order.currentStatusDate || order.createdDate).toISOString(), // XXX should we separate the dates?
-		blockchain_data: order.blockchainData!,
-		offer_type: order.type,
 		title: order.meta.title,
 		description: order.meta.description,
-		call_to_action: order.meta.call_to_action,
-		content: order.meta.content,
-		amount: order.amount,
+		amount: order.amount
 	};
+
+	return order.isMarketplaceOrder() ?
+		Object.assign(base, {
+			offer_type: order.type,
+			content: order.meta.content,
+			blockchain_data: order.blockchainData,
+			call_to_action: order.meta.call_to_action,
+		}) :
+		Object.assign(base, {
+			blockchain_data: {
+				recipient_address: order.meta.wallet_address
```

Comment:
order already has `blockchainData.recipient_address` why not use that instead of `order.meta`?

Reply:
3